### ISSUE TITLE
Return a 404 error for non existing routes

### ISSUE THREAD

**Issue by sudeeptoroy** [heptio/contour#982]
**What steps did you take and what happened:**
If an endpoint does not exists on nginx it returns an 404 not found error back. This functionality is missing on contour. On contour it silently fails by sending a blank response. 

**What did you expect to happen:**
return an 404 error back to the user.

**Anything else you would like to add:**
can you also provide a way to configure a default page for such missing routes (managed by platform folks).

**Environment:**

- Contour version: 0.10
- Kubernetes version: (use `kubectl version`): na
- Kubernetes installer & version: na
- Cloud provider or hardware configuration: na
- OS (e.g. from `/etc/os-release`): na

**Comment by stevesloka** [comment]
@sudeeptoroy Contour does return a 404 if the route isn't found if you check the http status code returned, however, there isn't a way to configure a default page to show.

**Comment by sudeeptoroy** [comment]
We have observed a small pause and blank response, however for nginx the error is displayed by curl/browser.

**Comment by davecheney** [comment]
Thank you for raising this issue. What I see enovy doing is return the correct status code but it does not return a body in the 40x or 50x response with a textual explanation of status code. This is not required for HTTP compliance, but that's an annoying nit picking technical response. 

Looking upstream this feature is in discussion for Envoy 0.11 (the next to next release, so maybe 3-4 months out), https://github.com/envoyproxy/envoy/issues/1178. I am marking this issue blocked until Envoy adds support for this feature.

**Comment by sudeeptoroy** [comment]
Sure, thanks

**Comment by davecheney** [comment]
Closing as duplicate of #320 

Waiting on upstream feature, we’ll follow up on #320 when this is available.
